src/ModelContextProtocol/McpChatClientBuilderExtensions.cs

```diff
@@ -0,0 +1,230 @@
+﻿using System.Collections.Concurrent;
+using System.Runtime.CompilerServices;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using ModelContextProtocol.Client;
+#pragma warning disable MEAI001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+namespace ModelContextProtocol;
+
+/// <summary>
+/// Extension methods for adding MCP client support to chat clients.
+/// </summary>
+public static class McpChatClientBuilderExtensions
+{
+    /// <summary>
+    /// Adds a chat client to the chat client pipeline that creates an <see cref="McpClient"/> for each <see cref="HostedMcpServerTool"/> 
+    /// in <see cref="ChatOptions.Tools"/> and augments it with the tools from MCP servers as <see cref="AIFunction"/> instances.
+    /// </summary>
+    /// <param name="builder">The <see cref="ChatClientBuilder"/> to configure.</param>
+    /// <param name="httpClient">The <see cref="HttpClient"/> to use, or <see langword="null"/> to create a new instance.</param>
+    /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use, or <see langword="null"/> to resolve from services.</param>
+    /// <returns>The <see cref="ChatClientBuilder"/> for method chaining.</returns>
+    /// <remarks>
+    /// <para>
+    /// When a <c>HostedMcpServerTool</c> is encountered in the tools collection, the client 
+    /// connects to the MCP server, retrieves available tools, and expands them into callable AI functions.
+    /// Connections are cached by server address to avoid redundant connections.
+    /// </para>
+    /// <para>
+    /// Use this method as an alternative when working with chat providers that don't have built-in support for hosted MCP servers.
+    /// </para>
+    /// </remarks>
+    public static ChatClientBuilder UseMcpClient(
```

Comment:
We should mark the public surface are for this as `[Experimental]`. We could do so either with "MEAI001", so that it matches with the MCP types on which its based (and in which case you wouldn't need the earlier suppression) or have some new "MCP003" or something.